tools/dhtnode.cpp

```diff
@@ -472,6 +510,15 @@ main(int argc, char **argv)
         }
 
         if (params.daemonize or params.service) {
+#if OPENDHT_PROXY_SERVER
+            if (params.proxyserver != 0) {
+                new DhtProxyServer(dht, params.proxyserver
```

Comment:
See comment above (also true for proxyclient). Here, the object lifetime is not correctly handled, so the server is not shutdown properly on exit.